src/map/script.c

```diff
@@ -6636,32 +6638,34 @@ BUILDIN(warpguild)
 		return true;
 	}
 
-	iter = mapit_getallusers();
-	for (pl_sd = BL_UCAST(BL_PC, mapit->first(iter)); mapit->exists(iter); pl_sd = BL_UCAST(BL_PC, mapit->next(iter))) {
-		if( pl_sd->status.guild_id != gid )
-			continue;
 
-		switch( type )
-		{
+	for (i = 0; i < MAX_GUILD; i++) {
+		if (g->member[i].online && g->member[i].sd != NULL) {
+
+			if (str2 != NULL && strcmp(str2, map->list[g->member[i].sd->bl.m].name) != 0)
```

Comment:
I think it'd be a better idea to avoid a string comparison on each member, but rather retrieve the map id (i.e. `map->mapname2mapid()` before the loop), and then compare just that to each member's `.sd->bl.m`